### PR TITLE
Apply "standard" linting style

### PR DESCRIPTION
**Pull request by ahmadnassri**
@espadrine as mentioned in #411 applied [standard](https://github.com/feross/standard) linting style which covers #276 and more.

this is not ready to be merged, I only applied automated linting and manually to the obvious fixes, I still have to compare the code it self and see why the contributors are sometime comparing a variable to itself!

making this PR to demo how it would look with standard syntax cleanup. if you :+1: I'll continue and make 100% pass.

edit: I see some undesired commits came through as well, ignore those, i'll rebase and re-open if you think this is going in a good direction.

### CODE REVIEW

File: lru-cache.js [hunk 1]
```diff
@@ -1,94 +1,99 @@
+'use strict'
+
 // Cache any data with a timestamp,
 // remove only the oldest data.
 
 var typeEnum = {
   unit: 0,
-  heap: 1,
```
Comment (espadrine) [inline lru-cache.js]: I definitely want to allow trailing commas in object and array literals.

**Comment by espadrine** [comment]
I definitely do want to keep the current implicit style guide, for the most part. Semicolons in particular help me (and others) not have to thing about accidental function calls (especially when using IIFE). Besides, it isn't much of a hassle.

However, apart from that, I'm actually more interested about linting (eg, knowing when a variable isn't used) than style (eg, knowing when a semicolon is missing), because one can hint at functional problems, while the other never will. Does it raise such issues somewhere? It'll be clearer with semistandard, I suppose.

Thanks for the work!

**Comment by ahmadnassri** [comment]
[standard](https://www.npmjs.com/package/standard) does **both** style & linting *(using eslint)*, however its [very opinionated](https://www.npmjs.com/package/standard#rules) and cant' customize rules... 

was hard to swallow at first, but having read all the research @feross did to come up with the final rules, I was satisfied and decided to offload all linting / style to *standard*.

that said, its not for everybody of course, and some rules might be annoying at times :)

and yes, *[semistandard](https://www.npmjs.com/package/semistandard)* is the same with semi-colons (happy to make that PR as well)

the purpose of this PR is just to show-case how comprehensive *standard* is (style & linting), if semistandard is a better choice for you, can do that as well :)

**Comment by espadrine** [comment]
It'd definitely be semistandard over standard. I don't have an issue with opinionated things if they avoid me shooting myself in the foot, and as long as they don't make me lose too much sleep.

I'm curious how severe the change would be on server.js.

**Comment by ahmadnassri** [comment]
I'll make more edits this weekend and update this PR.

commit history will be dirty, but we can squash / cherry pick before any
merge is needed.

**Comment by paulmelnikow** [comment]
We're handling linting gradually via #892 and #901 and beyond.

May be worth considering semistandard a target.

As a personal aside, I've stopped using semicolons. ASI is terrible. However, _it can't be turned off_. Placing the desired semicolons doesn't prevent ASI from inserting more bad ones. There are error cases of accidental insertion with _both styles_, which basically are a wash. You need a linter to avoid shooting yourself in the foot. Since semicolons are expensive – _a whole character for every line of code!_ – and _you need the linter anyway_, they're a clear loss. I prefer to omit the semicolons and lint unexpected multiline statements and unreachable code, which prevent the errors.

Sources:

- https://www.youtube.com/watch?v=Qlr-FGbhKaI
- http://blog.izs.me/post/2353458699/an-open-letter-to-javascript-leaders-regarding

**Comment by feross** [comment]
@paulmelnikow You're absolutely correct! I wrote about the need to understand ASI no matter what style you choose in a recent blog post: https://feross.org/never-use-semicolons/ It's something that often gets overlooked in this age old debate ;-)

**Comment by ahmadnassri** [comment]
closing in favor of referenced issues.

**Comment by espadrine** [comment]
I think of semicolons as clothes; one day you want to wear an orange T-shirt, the other a blue shirt. But wearing half of both would be strange, and I'd rather not invalidate every PR.

That said, I sometimes wish to see the parallel universe where Netscape used Scheme instead of JavaScript — then semicolons would be more than just style!

**Pull request closed without merging**